core/src/main/java/org/axonframework/commandhandling/model/AggregateMember.java

```diff
@@ -39,16 +42,41 @@
 
     /**
      * Indicates whether commands should be forwarded to this AggregateMember. Defaults to {@code true}.
+     *
+     * @return a {@code boolean} denoting iff commands should be forwarded, yes or no.
      */
     boolean forwardCommands() default true;
 
     /**
      * Indicates whether events should be forwarded to this AggregateMember. Defaults to {@code true}.
+     *
+     * @return a {@code boolean} denoting if events should be forwarded, yes or no.
+     *
+     * @deprecated in favor of the {@code eventForwardingMode}.
      */
+    @Deprecated
     boolean forwardEvents() default true;
 
+    /**
+     * Indicates the forwarding mode used for events within this entity. Defaults to {@code ForwardingMode.ALL} to allow
+     * all events through.
+     *
+     * @return a {@link org.axonframework.commandhandling.model.ForwardingMode} describing what mode should be used in
+     * forwarding events to this Aggregate Member.
+     */
+    ForwardingMode eventForwardingMode() default ForwardingMode.ALL;
+
+    /**
+     * The property of the event to be used as a routing key towards this Aggregate Member.
+     *
+     * @return The property of the event to use as routing key.
+     */
+    String eventRoutingKey() default "";
+
     /**
      * Provides the member's type. By default the type of member is determined from the field's generic type.
+     *
+     * @return a {@link java.lang.Class} instance of the type of this Aggregate Member
```

Comment:
Then we should remove the `@return` description in several places. 
I added because the javadoc of our `MessageHandler` annotations do have it (so added for consistency).

I'll however remove this addition. We can remove it from the message handler annotations somewhere in the future.

Reply:
I do see it was Rene who added those, so you're free from blame ;p